pkg/iac/adapters/terraform/aws/iam/convert.go

```diff
@@ -208,29 +199,28 @@ func parseStatement(statementBlock *terraform.Block) iamgo.Statement {
 	return builder.Build()
 }
 
-func findAllPolicies(modules terraform.Modules, parentBlock *terraform.Block, attr *terraform.Attribute) []wrappedDocument {
+func findAllPolicies(modules terraform.Modules, attr *terraform.Attribute) []wrappedDocument {
 	var documents []wrappedDocument
-	for _, ref := range attr.AllReferences() {
-		for _, b := range modules.GetBlocks() {
-			if b.Type() != "data" || b.TypeLabel() != "aws_iam_policy_document" {
-				continue
-			}
-			if ref.RefersTo(b.Reference()) {
-				document, err := ConvertTerraformDocument(modules, b)
-				if err != nil {
-					continue
-				}
+
+	if !attr.IsIterable() {
+		return documents
+	}
```

Comment:
Can we add a test case for this?

Reply:
Done https://github.com/aquasecurity/trivy/pull/6276/commits/9058054284df6732c1d114271cf0b02dfd07a031